src/gtc/dace/expansion/expansion.py

```diff
@@ -133,7 +133,14 @@ def expansion(
             node.oir_node, global_ctx=DaCeIRBuilder.GlobalContext(library_node=node, arrays=arrays)
         )
 
-        nsdfg = StencilComputationSDFGBuilder().visit(daceir)
+        nsdfg: dace.nodes.NestedSDFG = StencilComputationSDFGBuilder().visit(daceir)
+
+        delkeys = set()
+        for sym in node.symbol_mapping.keys():
+            if str(sym) not in nsdfg.free_symbols:
+                delkeys.add(str(sym))
+        for key in delkeys:
+            del node.symbol_mapping[key]
```

Comment:
The rest of this function is high level function calls. Would be clean if this is in a separate funtion with a descriptive name. :)

Reply:
I moved it to the above `_fix_context` which I think is a suitable place, and added a comment.